src/components/Table/index.js

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import resolveColumns from './resolve-columns';
+import Rows from './body';
+import Headers from './head';
+import './styles.css';
+
+/** Data tables display information in a way that’s easy to scan,
+ * so that users can look for patterns and insights. */
+export default class Table extends Component {
+    constructor(props) {
+        super(props);
+        this.columns = undefined;
+    }
+
+    componentDidMount() {
+        if (!this.columns) {
+            this.getColumns();
```

Comment:
I think if the columns is always generated from the column passed by the developer and the change of the column imply render of the table then we should put it in the state of the component

Reply:
@reiniergs but setting it in the `state` would cause an extra render cycle and as it`s pretty much a read-only property, it'll work fine as a class property